### PR TITLE
[IMP] stock: 2654703

### PR DESCRIPTION
**Pull request by boan-odoo**
[IMP] stock: resize prod. form view fields

task: 2654703-9
[IMP] stock: show related products on duplicated barcode

task: 2654703-7
[IMP] stock: recolor operations status badges

task: 2654703-4
[IMP] stock: add batch transfer column to moves view

task: 2654703-2-3
[IMP] stock: group product moves by carrier

task: 2654703-2-2
[IMP] stock: group product moves by category

task: 2654703-2-1

### CODE REVIEW

File: addons/product/i18n/product.pot [hunk 1]
```diff
@@ -2598,6 +2593,15 @@ msgstr ""
 msgid "There are no remaining possible combination."
 msgstr ""
 
+#. module: product
```
Comment (amoyaux) [inline addons/product/i18n/product.pot:2596]: no need for .pot in master

Comment (amoyaux) [reply addons/product/i18n/product.pot:2596]: You can remove it

File: addons/product/models/product.py [hunk 2]
```diff
@@ -227,9 +228,34 @@ def init(self):
...
+
+        # Collect duplicates (if there are any)
+        duplicates = defaultdict(set)
+        for product in self:
+
+            # Skip check on products without a barcode
+            if not product.barcode:
+                continue
+
+            products_with_same_barcode = self.env['product.product'].sudo().search(
```
Comment (hparfr) [inline addons/product/models/product.py:242]: Please take care to not introduce performance issues.

Comment (boan-odoo) [reply addons/product/models/product.py:242]: I understand your concern. Do you have any recommendation regarding a different course of action that would achieve the same end goal without the additional performance impact?

Comment (hparfr) [reply addons/product/models/product.py]: IMO a good design can be: Allow inconsistencies in products (so allow barcode duplicates) and put some validation logic elsewhere, like do not allow to add a product in sale order if it's not compliant to some rules. 
An exception can be thrown at runtime or by a cron (exception like in stock for "procurement exceptions" / bad routing).

Plus here the barcode is a bit special: it's really opinionated. Only one barcode per product, unicity across all the companies, and there is edge cases when (de)archiving products. 


Back to the perf: I do not know if your code introduce a perf issue. I suspect it but I'm not an ORM specialist. My request is, before merging it, please test if there is no regressions against a db with much more than 10 products. :roll_eyes:

Comment (boan-odoo) [reply addons/product/models/product.py]: I see. I will talk about this with the rest of the team. Thank you very much for your input!

**Comment by amoyaux** [comment]
robodoo r+ rebase-ff

**Pull request closed without merging**